libs/SalesforceSwiftSDK/SalesforceSwiftSDK/Extensions/SFRestAPIExtensions.swift

```diff
@@ -26,13 +26,56 @@ import Foundation
 import PromiseKit
 import SalesforceSDKCore
 
+public struct SFRestResponse {
```

Comment:
Should that object belong in the objective-c SDK ?

Reply:
@wmathurin The wrapper helps with the swift-promise styled api(s) only. The obj-c restapi  completion blocks are defined with both the deserialized data and headers in the response.

Reply:
(Probably too late for 6.0) but maybe the completion block should also use an object like SFRestResponse. That would line up the implementation with Android and also simplifies some of the response parsing work we do in downstream consumers of SFRestAPI (see SFNetworkPlugin.m and SFNetReactBridge.m).